src/platform-includes/performance/distributed-tracing/java.mdx

```diff
@@ -1,32 +1,55 @@
-To obtain a trace header from the span, use `ISpan#toSentryTrace()` method. Then pass it to the downstream service. If the communication happens over HTTP, it is recommended that you set the value to the `sentry-trace` (also available as a constant `SentryTraceHeader.SENTRY_TRACE_HEADER`) HTTP header.
+To obtain a trace header from the span, use `ISpan#toSentryTrace()` method. Then pass it to the downstream service. If the communication happens over HTTP, it is recommended that you set the value to the `sentry-trace` (also available as a constant `SentryTraceHeader.SENTRY_TRACE_HEADER`) HTTP header as well as attaching the `baggage` header (`BaggageHeader.BAGGAGE_HEADER`) which can be retrieved from `ISpan#toBaggageHeader(List<String>)`. Please pass in any existing `baggage` header entries that are present on the outgoing request so we can handle duplicate Sentry entries and limits for you.
 
-To create a span as a continuation of the trace retrieved from the upstream service, pass the `sentry-trace` header value to the transaction context:
+To create a span as a continuation of the trace retrieved from the upstream service, pass the `sentry-trace` and `baggage` header values to the transaction context:
 
 ```java
+import io.sentry.Baggage;
+import io.sentry.BaggageHeader;
 import io.sentry.ITransaction;
 import io.sentry.Sentry;
 import io.sentry.SentryTraceHeader;
 import io.sentry.TransactionContext;
 import io.sentry.exception.InvalidSentryTraceHeaderException;
 
 String sentryTrace = request.getHeader(SentryTraceHeader.SENTRY_TRACE_HEADER);
+List<String> baggageHeaders = Collections.list(request.getHeaders(BaggageHeader.BAGGAGE_HEADER);
 ITransaction transaction = null;
 try {
-  transaction = Sentry.startTransaction(TransactionContext.fromSentryTrace("name", "op", new SentryTraceHeader(sentryTrace)));
+  transaction = Sentry.startTransaction(
+    TransactionContext.fromSentryTrace(
+      "name",
+      TransactionNameSource.CUSTOM,
+      "op",
+      new SentryTraceHeader(sentryTrace),
+      Baggage.fromHeader(baggageHeaders, hub.getOptions().getLogger())
```

Comment:
We need to expose a method `fromHeader` that doesn't take a `logger`.
`Sentry.getCurrentHub()` is `@ApiStatus.Internal` so people cannot use this.

Reply:
Or make it nullable, but I'd rather expose a new overload.

Reply:
How about passing in the options instead?

Reply:
Hmm so we're down to calling `Sentry.getCurrentHub().getOptions().getLogger()` inside then to hide it from the user but still be able to log problems I presume.

Reply:
That's fine, that's what `HubAdapter.getInstance()` stands for, at least to make it testable when calling internally.
I'd make a PR asap because otherwise, it's not usable, I don't expect people using `Baggage.fromHeader` themselves any way, so it's not a p0 but create an issue, and don't merge this before doing it, this would just raise more questions around "how do I get the logger?".